src/main/java/com/google/sps/servlets/NotificationServlet.java

```diff
@@ -0,0 +1,209 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+ 
+package com.google.sps.servlets;
+ 
+import com.google.sps.classes.Notification;
+import com.google.sps.classes.Utility;
+import java.io.IOException;
+import java.sql.*;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+ 
+/**
+* Servlet that handles the fetching and posting of notifications.
+*/
+@WebServlet("/notification")
+public class NotificationServlet extends HttpServlet {
+  /**
+  * Fetch the notifications of a specific user.
+  */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get user's ID from Utility method.
+    int userId = Utility.getUserId();
+    // If the user is signed in then fetch notifications.
+    if (userId != -1) {
+      // If the user is logged in, fetch notifications and convert the ArrayList to JSON
+      // using Utility method.
+      String json = Utility.convertToJsonUsingGson(getNotifications(userId));
+      response.setContentType("application/json;");
+      response.getWriter().println(json); 
+    }
+  }
+
+  /**
+  * Post a new notification for all followers of a modified question or answer.
+  */
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Define local time for the new entries in the server.
+    LocalDateTime localDateTime = LocalDateTime.now();
+    Timestamp localTimestamp = Timestamp.valueOf(localDateTime);
+
+    // Get if notification is question answered or answered commented and the id of the element.
+    String typeOfNotification = request.getParameter("type");
+    int modifiedElementId = Integer.parseInt(request.getParameter("modifiedElementId"));
+    
+    if(typeOfNotification.equals("question")) {
+      // If the notification is for an anwer to a question.
+      String query =  "SELECT follower_id FROM QuestionFollower WHERE question_id = " +
+                       modifiedElementId;
+      // Query the information from QuestionFollower table.
+      try (Connection connection = DriverManager.getConnection(Utility.SQL_LOCAL_URL, 
+                                                               Utility.SQL_LOCAL_USER,
+                                                               Utility.SQL_LOCAL_PASSWORD);
+          PreparedStatement pst = connection.prepareStatement(query);
+          ResultSet rs = pst.executeQuery()) {
```

Comment:
nit: Indent back two spaces.

Reply:
Indented it forward one space to align with connection.

Reply:
Oh, I just realized you made these lines part of the parenthesis, so yeah, you could just leave the four spaces or align with `connection`.